src/main/kotlin/app/revanced/patches/youtube/ad/general/resource/patch/GeneralResourceAdsPatch.kt

```diff
@@ -0,0 +1,60 @@
+package app.revanced.patches.youtube.ad.general.resource.patch
+
+import app.revanced.extensions.doRecursively
+import app.revanced.extensions.startsWithAny
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patches.youtube.ad.general.annotation.GeneralAdsCompatibility
+import app.revanced.patches.youtube.misc.manifest.patch.FixLocaleConfigErrorPatch
+import org.w3c.dom.Element
+
+@DependsOn(dependencies = [FixLocaleConfigErrorPatch::class])
+@Name("general-resource-ads")
+@Description("Patch to remove general ads in resources.")
+@GeneralAdsCompatibility
+@Version("0.0.1")
+class GeneralResourceAdsPatch : ResourcePatch {
+    // list of resource file names which need to be hidden
+    private val resourceFileNames = arrayOf(
+        "compact_promoted_video_item.xml",
```

Comment:
Are all those resources required to be patched?

Reply:
Removed unnecessary resources.